### PR TITLE
[MODEL] Add scripts for Joint intent classification and slot labelling with BERT

### PR DESCRIPTION
**Pull request by sxjscience**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [x] Intent classification and slot labelling tutorial

### CODE REVIEW

File: scripts/intent_slot/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Joint Intent Classification and Slot Labelling by GluonNLP
```
Comment (eric-haibin-lin) [inline scripts/intent_slot/README.md]: by GluonNLP -> with GluonNLP

File: scripts/intent_slot/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+| to        | O                   |  |
+| columbus  | B-toloc.city_name   |  |
+| on        | O                   |  |
+| wednesday | B-depart_date.day_name    |  |
+| april     | B-depart_date.month_name  |  |
+| sixty     | B-depart_date.day_number  |  |
+
+
+
+In this example, we demonstrate how to use GluonNLP to build a model to perform joint intent classification and slot labelling. We
```
Comment (eric-haibin-lin) [inline scripts/intent_slot/README.md]: Maybe add a reference link to what BIO stands for?

File: scripts/intent_slot/bert_icsl.ipynb [hunk 3]
```diff
@@ -0,0 +1,935 @@
+{
```
Comment (eric-haibin-lin) [inline scripts/intent_slot/bert_icsl.ipynb]: Notebooks are usually committed in the form of `.md` files like https://github.com/dmlc/gluon-nlp/blob/master/docs/examples/machine_translation/gnmt.md which is converted with notedown: https://gluon-nlp.mxnet.io/community/contribute.html#contribute-tutorials

File: scripts/intent_slot/bert_icsl.ipynb [hunk 4]
```diff
@@ -0,0 +1,935 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "batch_size = 32\n",
+    "learning_rate = 5E-5\n",
+    "\n",
+    "trainer = gluon.Trainer(net.collect_params(), 'bertadam',\n",
+    "                        {'learning_rate': learning_rate, 'wd': 0.0})\n",
+    "batchify_fn = nlp.data.batchify.Tuple(nlp.data.batchify.Pad(),    # Subword ID\n",
```
Comment (eric-haibin-lin) [inline scripts/intent_slot/bert_icsl.ipynb]: can you fix the Padding warning?

File: scripts/intent_slot/bert_icsl.ipynb [hunk 5]
```diff
@@ -0,0 +1,935 @@
...
+     "text": [
+      "[Epoch 4]    dev intent/slot = 0.118/0.092, slot f1 = 95.20, intent acc = 98.20\n",
+      "[Epoch 4]    test intent/slot = 0.127/0.146, slot f1 = 92.87, intent acc = 98.10\n"
+     ]
+    }
+   ],
+   "source": [
+    "from seqeval.metrics import f1_score as ner_f1_score\n",
+    "\n",
+    "def evaluation(ctx, data_loader, net, intent_pred_loss, slot_pred_loss, slot_vocab):\n",
```
Comment (eric-haibin-lin) [inline scripts/intent_slot/bert_icsl.ipynb]: The evaluation function is bulky. Can we break it down to even smaller pieces?

File: scripts/intent_slot/demo.py [hunk 6]
```diff
@@ -0,0 +1,399 @@
+import os, sys
```
Comment (eric-haibin-lin) [inline scripts/intent_slot/demo.py]: can we add a test in test_script.py ?

**Comment by mli** [comment]
Job PR-817/12 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/12/index.html

**Comment by mli** [comment]
Job PR-817/13 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/13/index.html

**Comment by mli** [comment]
Job PR-817/14 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/14/index.html

**Comment by mli** [comment]
Job PR-817/15 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/15/index.html

**Comment by mli** [comment]
Job PR-817/16 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/16/index.html

**Comment by mli** [comment]
Job PR-817/17 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/17/index.html

**Comment by mli** [comment]
Job PR-817/18 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-817/18/index.html
